app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -74,6 +91,60 @@ def save_inventory_multi(association, hashes, deletes, find_key, child_keys = []
     association.push(new_records)
   end
 
+  # Saves a inventory record, and any sub inventory records, without
+  # associating it with the parent object that is passed in.  Parent usually is
+  # an EMS, but doesn't have to be.
+  #
+  # @param parent [ActiveRecord] base record being saved
+  # @param association [Symbol]  associated record type being added to parent
+  # @param hashes [Array]        inventory data (hash per record)
+  # @param find_key [Array]      ordered keys for existing record lookup
+  # @param child_keys [Array]    keys from hashes that are child records
+  # @param extra_keys [Array]    hash keys to be removed before saving records
+  # @param disconnect [Boolean]  whether or not to delete or disconnect records
+  #                              that don't exist in the inventory hashes, but
+  #                              are present in the DB
+  #
+  # @return [Hash] the hashes passed into the method
+  def save_inventory_with_thin_association(parent, association, hashes, find_key,
+                                           child_keys = [], extra_keys = [],
+                                           disconnect = false)
+
+    relation      = parent.association(association).reflection
+    model         = parent.association(association).klass
+    assoc_attrs   = parent.association(association).reader.scope_attributes
```

Comment:
hm, this seems to be returning also :type, so that might break our STI models?

also it sends "type" as a list of allowed STI types, I think think that is being ignored by AR since we also provider :type after that

Reply:
Good catch.  I will see what `ActiveRecord` does under the hood with this one to see if that will be an issue.  This is what I pulled from when looking into how `.build` was used, but I forget the specifics now that it has been about two weeks since I got this put together.

I will do some digging and report back.